src/backend/cdb/cdbutil.c

```diff
@@ -952,6 +957,45 @@ cdbcomponent_activeQEsExist(void)
 	return !cdb_component_dbs ? false : cdb_component_dbs->numActiveQEs > 0;
 }
 
+bool
+cdb_checkWarehouseName(char *new_name)
+{
+	if (Gp_role == GP_ROLE_EXECUTE)
+		return true;
+	if (new_name == NULL || strcmp(new_name, "default") == 0)
+		return true;
+
+	Relation	rel = table_open(GpSegmentConfigRelationId, AccessShareLock);
+	HeapTuple	tuple;
+	SysScanDesc sscan;
+	bool		warehouse_exist = false;
+	bool		isNull;
+	Datum		attr;
+
+	sscan = systable_beginscan(rel, InvalidOid, false, NULL, 0, NULL);
+	while ((tuple = systable_getnext(sscan)) != NULL)
+	{
+		char *warehouse_name = NULL;
+		attr = heap_getattr(tuple, Anum_gp_segment_configuration_warehouse_name,
+							RelationGetDescr(rel), &isNull);
+		if (!isNull)
+			warehouse_name = TextDatumGetCString(attr);
+
+		if (warehouse_name && strcmp(warehouse_name, new_name) == 0)
```

Comment:
Is it possible drop warehouse?

Reply:
We need to process if warehouse is used by another session, we cannot drop that warehouse by another session.